lib/mapObjects/CGTownInstance.cpp

```diff
@@ -46,7 +46,7 @@ void CCreGenAsCastleInfo::serializeJson(JsonSerializeFormat & handler)
 
 	if(!handler.saving)
 	{
-		asCastle = (instanceId != "");
+		asCastle = (!instanceId.empty());
```

Comment:
```suggestion
		asCastle = !instanceId.empty();
```

Reply:
@kambala-decapitator fix it here https://github.com/vcmi/vcmi/pull/1111/commits/2c01e9680c3778d5f4a2732eb6f7d01c42fc2501